test/StringUtils.test.cpp

```diff
@@ -1,8 +1,370 @@
+	EXPECT_EQ(0, NAS2D::stringTo<long long>("0"));
+	EXPECT_EQ(1, NAS2D::stringTo<long long>("1"));
+	EXPECT_THROW(NAS2D::stringTo<long long>(""), std::invalid_argument);
+	EXPECT_THROW(NAS2D::stringTo<long long>(bigNum), std::out_of_range);
+
+	EXPECT_THROW(NAS2D::stringTo<float>("-" + bigNum), std::out_of_range);
+	EXPECT_EQ(-1.0, NAS2D::stringTo<float>("-1"));
+	EXPECT_EQ(-1.0, NAS2D::stringTo<float>("-1.0"));
+	EXPECT_EQ(0.0, NAS2D::stringTo<float>("0"));
+	EXPECT_EQ(0.0, NAS2D::stringTo<float>("0.0"));
+	EXPECT_EQ(1.0, NAS2D::stringTo<float>("1"));
+	EXPECT_EQ(1.0, NAS2D::stringTo<float>("1.0"));
+	EXPECT_EQ(1.1f, NAS2D::stringTo<float>("1.1")); // Subject to rounding
+	EXPECT_THROW(NAS2D::stringTo<float>(""), std::invalid_argument);
+	EXPECT_THROW(NAS2D::stringTo<float>(bigNum), std::out_of_range);
+
+	EXPECT_EQ(-1.0, NAS2D::stringTo<double>("-1"));
+	EXPECT_EQ(-1.0, NAS2D::stringTo<double>("-1.0"));
+	EXPECT_EQ(0.0, NAS2D::stringTo<double>("0"));
+	EXPECT_EQ(0.0, NAS2D::stringTo<double>("0.0"));
+	EXPECT_EQ(1.0, NAS2D::stringTo<double>("1"));
+	EXPECT_EQ(1.0, NAS2D::stringTo<double>("1.0"));
+	EXPECT_EQ(1.1, NAS2D::stringTo<double>("1.1")); // Subject to rounding
+	EXPECT_THROW(NAS2D::stringTo<double>(""), std::invalid_argument);
+
+	EXPECT_EQ(-1.0, NAS2D::stringTo<long double>("-1"));
+	EXPECT_EQ(-1.0, NAS2D::stringTo<long double>("-1.0"));
+	EXPECT_EQ(0.0, NAS2D::stringTo<long double>("0"));
+	EXPECT_EQ(0.0, NAS2D::stringTo<long double>("0.0"));
+	EXPECT_EQ(1.0, NAS2D::stringTo<long double>("1"));
+	EXPECT_EQ(1.0, NAS2D::stringTo<long double>("1.0"));
+	EXPECT_EQ(1.1L, NAS2D::stringTo<long double>("1.1")); // Subject to rounding
+	EXPECT_THROW(NAS2D::stringTo<long double>(""), std::invalid_argument);
+}
+
+TEST(String, stringFromStdString)
+{
+	using namespace std::literals::string_literals;
+
+	EXPECT_EQ(""s, NAS2D::stringFrom<std::string>(""));
+	EXPECT_EQ("Hello World"s, NAS2D::stringFrom<std::string>("Hello World"));
+}
+
+
+TEST(String, stringFromBool)
+{
+	using namespace std::literals::string_literals;
+
+	EXPECT_EQ(std::string{""}, NAS2D::stringFrom<std::string>(""));
+	EXPECT_EQ(std::string{"Hello World"}, NAS2D::stringFrom<std::string>("Hello World"));
+
+	EXPECT_EQ("true"s, NAS2D::stringFrom<bool>(true));
+	EXPECT_EQ("false"s, NAS2D::stringFrom<bool>(false));
+}
+
+TEST(String, stringFromChar)
+{
+	using namespace std::literals::string_literals;
+
+	EXPECT_EQ("A"s, NAS2D::stringFrom<char>('A'));
+	EXPECT_EQ("B"s, NAS2D::stringFrom<char>('B'));
+	EXPECT_EQ("0"s, NAS2D::stringFrom<char>('0'));
+	EXPECT_EQ("1"s, NAS2D::stringFrom<char>('1'));
+	EXPECT_EQ("a"s, NAS2D::stringFrom<char>('a'));
+	EXPECT_EQ("b"s, NAS2D::stringFrom<char>('b'));
+	EXPECT_EQ("129"s, NAS2D::stringFrom<unsigned char>(129));
+	EXPECT_EQ("-1"s, NAS2D::stringFrom<signed char>(-1));
+
+}
+
+#if defined(__clang__)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wgnu-zero-variadic-macro-arguments"
+#endif
```

Comment:
We can probably reduce the scope of this block. I think it might only need to wrap the use of `TYPED_TEST_SUITE`.

We can potentially combine the code for multiple types into one region. Looks like they can all be declared up front too. Maybe we should set a generic type alias at the head of the file for `SignedTypes` and `UnsignedTypes`, and then use the macro up near the top. The test methods can be below. It looks like that ordering is already used, so it should work.